test/AaveAtokenFork.t.sol

```diff
@@ -0,0 +1,330 @@
+    // reserved for the mapping (i.e. 52, as found with `forge inspect`), then hash
+    // the result. *Then* we can cast and add the offset to find the position of
+    // the 5th slot for the struct.
+    //
+    // This is all per:
+    // https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html
+    bytes32 _aTokenAddressStorageSlot = bytes32(uint256(keccak256(
+      bytes.concat(
+        bytes32(uint256(uint160(address(govToken)))), // map key == the govToken addr
+        bytes32(uint256(52)) // _reserves slot, as determined by forge
+      )
+    )) + 4); // 4 slots *after* the slot computed for the struct, i.e. the 5th slot
+
+    aToken = AToken(
+      address(uint160(uint256(
+        vm.load(address(pool), _aTokenAddressStorageSlot)
+      )))
+    );
+
+    // Sometimes Aave uses oracles to get price information, e.g. when
+    // determining the value of collateral relative to loan value. Since GOV
+    // isn't a real thing and doesn't have a real price, we need to mock these
+    // calls. When borrowing, the oracle interaction happens in
+    // GenericLogic.calculateUserAccountData L130
+    address _priceOracle = pool.ADDRESSES_PROVIDER().getPriceOracle();
+    vm.mockCall(
+      _priceOracle,
+      abi.encodeWithSelector(
+        AaveOracle.getAssetPrice.selector,
+        address(govToken)
+      ),
+      // Aave only seems to use USD-based oracles, so we will do the same.
+      abi.encode(1e8) // 1 GOV == $1 USD
+    );
+
+  }
+
+  function testFork_Setup_CanSupplyGovToAave() public {
+    assertEq(ERC20(address(aToken)).symbol(), "aOptGOV");
+    assertEq(ERC20(address(aToken)).name(), "Aave V3 Optimism GOV");
+
+    // Confirm that the atoken._underlyingAsset == govToken
+    //
+    //   $ forge inspect lib/aave-v3-core/contracts/protocol/tokenization/AToken.sol:AToken storage
+    //     ...
+    //     "label": "_underlyingAsset",
+    //     "offset": 0,
+    //     "slot": "61",
+    //     "type": "t_address"
+    assertEq(
+      address(uint160(uint256(
+        vm.load(address(aToken), bytes32(uint256(61)))
+      ))),
+      address(govToken)
+    );
+
+    // Mint GOV and deposit into aave.
+    // Confirm that we can supply GOV to the aToken.
+    assertEq(aToken.balanceOf(address(this)), 0);
+    govToken.THIS_IS_JUST_A_TEST_HOOK_mint(address(this), 42 ether);
+    govToken.approve(address(pool), type(uint256).max);
+    pool.supply(
+      address(govToken),
+      2 ether,
+      address(this),
+      0 // referral code
+    );
+    assertEq(govToken.balanceOf(address(this)), 40 ether);
+    assertEq(aToken.balanceOf(address(this)), 2 ether);
+
+    // We can withdraw our GOV when we want to.
+    pool.withdraw(
+      address(govToken),
+      2 ether,
+      address(this)
+    );
+    assertEq(govToken.balanceOf(address(this)), 42 ether);
+    assertEq(aToken.balanceOf(address(this)), 0 ether);
+  }
+
+  function testFork_Setup_CanBorrowAgainstGovCollateral() public {
```

Comment:
Similarly, per that doc, this should be something like `testFork_CanBorrowAgainstGovCollateral`. I'm not sure what the `Setup` part means, maybe that this is testing post-setup state as opposed to actually testing our code? If so we can just adjust my suggestion accordingly

Reply:
I think(?) I did what you want in 8a07568

Reply:
LGTM, thanks!